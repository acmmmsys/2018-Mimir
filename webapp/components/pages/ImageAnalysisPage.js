import React from 'react'

import ContentGrid from 'layout/ContentGrid';
import ContentColumn from 'layout/ContentColumn';
import HeaderGrid from 'layout/HeaderGrid';
import HeaderColumn from 'layout/HeaderColumn';
import ContentRow from 'layout/ContentRow';

import ButtonUpload from 'components/button/ButtonUpload';
import ButtonToggleAttachFile from 'components/button/ButtonToggleAttachFile';

import Panel from 'layout/Panel'
import PanelHeader from 'layout/PanelHeader';

import PanelShowcaseSelectedFile from 'components/panels/PanelShowcaseSelectedFile';
import PanelGuidedGradCamVisualization from 'components/panels/PanelGuidedGradCamVisualization'
import PanelGradCamVisualization from 'components/panels/PanelGradCamVisualization'
import PanelLayerSelection from 'components/panels/PanelLayerSelection'
import PanelTargetSelection from 'components/panels/PanelTargetSelection'
import PanelImageDescription from 'components/panels/PanelImageDescription';
import PanelFileSelector from 'components/panels/PanelFileSelector';

const ImageAnalysisPage = () => 
    <div className="full">
        <HeaderGrid>
            <HeaderColumn>
                <PanelHeader>
                    <ButtonToggleAttachFile />
                </PanelHeader>
            </HeaderColumn>
            <HeaderColumn>
                <PanelHeader>
                    <ButtonUpload />
                </PanelHeader>
            </HeaderColumn>
        </HeaderGrid>
        <ContentGrid>
            <ContentColumn>
                <ContentRow>
                    <ContentColumn>
                        <PanelShowcaseSelectedFile />
                    </ContentColumn>
                    <ContentColumn>
                        <PanelImageDescription />
                    </ContentColumn>
                </ContentRow>
                <ContentRow>
                    <ContentColumn>
                        <PanelGuidedGradCamVisualization />
                    </ContentColumn>
                    <ContentColumn>
                        <PanelGradCamVisualization />
                    </ContentColumn>
                </ContentRow>
                <ContentRow>
                    <ContentColumn>
                        <PanelLayerSelection />
                    </ContentColumn>
                    <ContentColumn>
                        <PanelTargetSelection />
                    </ContentColumn>
                </ContentRow>
            </ContentColumn>
            <ContentColumn>
                <PanelFileSelector />
            </ContentColumn>
        </ContentGrid>
    </div>

export default ImageAnalysisPage;