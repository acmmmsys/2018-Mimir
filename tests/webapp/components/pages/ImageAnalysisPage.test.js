import React from 'react';
import ImageAnalysisPage from '../../../../webapp/components/pages/ImageAnalysisPage'
import renderer from 'react-test-renderer';

let wrapper;

test('ImageAnalysisPage does render', () => {
    wrapper = shallow(
        <ImageAnalysisPage>

        </ImageAnalysisPage>
    );
    expect(wrapper).toMatchSnapshot();
});
