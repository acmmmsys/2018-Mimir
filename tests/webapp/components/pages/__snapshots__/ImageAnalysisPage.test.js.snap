// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageAnalysisPage does render 1`] = `
<div
  className="full"
>
  <HeaderGrid>
    <HeaderColumn>
      <defaultProps(branch(branch(PanelHeader)))
        emptyMessage=""
        loading={false}
      >
        <Connect(branch(ButtonAction)) />
      </defaultProps(branch(branch(PanelHeader)))>
    </HeaderColumn>
    <HeaderColumn>
      <defaultProps(branch(branch(PanelHeader)))
        emptyMessage=""
        loading={false}
      >
        <Connect(ButtonUpload) />
      </defaultProps(branch(branch(PanelHeader)))>
    </HeaderColumn>
  </HeaderGrid>
  <ContentGrid>
    <ContentColumn>
      <ContentRow>
        <ContentColumn>
          <Connect(withProps(nest(defaultProps(branch(branch(Panel))), ShowcaseImage))) />
        </ContentColumn>
        <ContentColumn>
          <Connect(withProps(nest(defaultProps(branch(branch(Panel))), ImageDescription))) />
        </ContentColumn>
      </ContentRow>
      <ContentRow>
        <ContentColumn>
          <Connect(withProps(nest(defaultProps(branch(branch(Panel))), ShowcaseImage))) />
        </ContentColumn>
        <ContentColumn>
          <Connect(withProps(nest(defaultProps(branch(branch(Panel))), ShowcaseImage))) />
        </ContentColumn>
      </ContentRow>
      <ContentRow>
        <ContentColumn>
          <Connect(withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector))))) />
        </ContentColumn>
        <ContentColumn>
          <Connect(defaultProps(withHandlers(withProps(lifecycle(nest(defaultProps(branch(branch(Panel))), TableSelector)))))) />
        </ContentColumn>
      </ContentRow>
    </ContentColumn>
    <ContentColumn>
      <Connect(lifecycle(withProps(nest(defaultProps(branch(branch(Panel))), FileSelector)))) />
    </ContentColumn>
  </ContentGrid>
</div>
`;
